### ISSUE TITLE
`missing-docs-allow-unused` doesn't catch associated `const`s.

### ISSUE THREAD

**Issue by Qix-** [rust-lang/rust-clippy#17052]
### Summary

I suggested `missing-docs-allow-unused` in #14413 which has been useful, but it seems it was a bit constrained in its scope and only applies to fields. However I ran into the case today where the `missing-docs-allow-unused` behaved unexpectedly - it didn't silence an error in a `const _SIZE_TEST: () = { assert!(...) }` declaration where I'd have expected it to.

Could it be expanded to include any declaration that has an identifier that is `_`-prefixed?

### Lint Name

missing_docs_in_private_items

### Reproducer

I tried this code:

```rust
//! Test
pub struct Foo(pub usize);

impl Foo {
    const _SIZE_ASSERT: () = {
        assert!(
            core::mem::size_of::<Self>().is_multiple_of(4),
            "not a multiple of 4"
        );
    };
}
```

```toml
# clippy.toml
missing-docs-allow-unused = true
```

I saw this happen:

```
error: missing documentation for an associated constant
  --> src/lib.rs:5:5
   |
 5 | /     const _SIZE_ASSERT: () = {
 6 | |         assert!(
 7 | |             core::mem::size_of::<Self>().is_multiple_of(4),
 8 | |             "not a multiple of 4"
 9 | |         );
10 | |     };
   | |______^
```

I expected no clippy lint to emit.

### Version

```text
clippy 0.1.90 (1159e78c47 2025-09-14)
```
